### ISSUE TITLE
Make UidReverseComparator transitive

### ISSUE THREAD

**Issue by eckst** [thunderbird/thunderbird-android#969]
After loading a new mail message from pop3 server all messages dissapeared and in the inbox was visible just the message IllegalArgumentException. All messages were lost. I reinstalled K9, downloaded messages from pop3 server again, but two days later the same error arose and messages were lost again. 
I encountered this bug a year ago for the first time but now it's here again.

**Comment by cketti** [comment]
Please provide a [debug log](https://github.com/k9mail/k-9/wiki/LoggingErrors).
Also, what K-9 Mail version are you using?

**Comment by eckst** [comment]
K9-Mail version: 5.007

I've installed K9 again, since I removed it from the mobile phone already. I downloaded a few mails again and a different error IllegalArgumentException appeared, but those downloaded messages were not deleted this time. In the error message folder were some messages, this is one of them:

IllegalArgumentException: Comparison method violates its general contract!

K9-Mail version: 5.007Device make: AcerDevice model: E380Android version: 4.4.2java.lang.IllegalArgumentException: Comparison method violates its general contract!	at java.util.TimSort.mergeHi(TimSort.java:864)	at java.util.TimSort.mergeAt(TimSort.java:481)	at java.util.TimSort.mergeCollapse(TimSort.java:404)	at java.util.TimSort.sort(TimSort.java:210)	at java.util.TimSort.sort(TimSort.java:169)	at java.util.Arrays.sort(Arrays.java:2010)	at java.util.Collections.sort(Collections.java:1883)	at com.fsck.k9.controller.MessagingController.downloadMessages(MessagingController.java:1311)	at com.fsck.k9.controller.MessagingController.synchronizeMailboxSynchronous(MessagingController.java:1141)	at com.fsck.k9.controller.MessagingController.access$400(MessagingController.java:111)	at com.fsck.k9.controller.MessagingController$8.run(MessagingController.java:934)	at
com.fsck.k9.controller.MessagingController.run(MessagingController.java:435)	at java.lang.Thread.run(Thread.java:841)

I've removed the K9 again.
